Fix failing tests

sculpt0r opened the pull request:
First of all, I started with defensive `if` on the fragile `CKEDITOR` in the observer (main repository). However, not all tests passed.

The most damaged makes setup editor from CDN via `editorUrl` props. So I moved it to the last position, so no other tests will fail because of it: https://github.com/ckeditor/ckeditor4-vue/pull/127/commits/844da89a3c6c74bf55e939f300b737e087202ea5

I have a problem with not incoming events from the main editor code, like `ready` and I decided that I change an assertion, so it shows if the event was invoked at all, rather than verify only if it was one time. Because no calls of the spy in the test - means that some event wasn't raised at a deep level, whereas multiple calls tell us about problems in integration commons:  https://github.com/ckeditor/ckeditor4-vue/pull/127/commits/c3fc2a18013ccda70043e80336b00e1f109d6333

Based debug in karma run, I found that the failing test `should use correct CKEDITOR build` try to load some assets from the different origin that it was set via `editorUrl`. And it appears that during the test we are deleting `window.CKEDITOR` but there are still scripts added via 'namespaceloader'. So I decided to remove all `ckeditor.js` scripts-like along with deleting global `window.CKEDITOR`. That fixes the last test assertion problem: https://github.com/ckeditor/ckeditor4-vue/pull/127/commits/fdb6de33630f64aa41dd2d9eaee4aaae30b45444 However, I'm still thinking, that if you load CKE4 from the CDN via `editorUrl` and then, you check `editorInstance.config.basePath` - it's different. It has good parts, but that may be something worth checking with https://github.com/ckeditor/ckeditor4/issues/4761.

I was able to fix the issue with `MutationObserver` in vue tests, thanks to also new config option: https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-observableParent It tells the observer, which node should be checked and in case of changes - determine if there is an CKE4 instance inside. If we provide an immutable element, then observer callback will never be run: https://github.com/ckeditor/ckeditor4-vue/pull/127/commits/a7450096434116fd60324d8528ac615920fb676d

Finally, I have to adjust the code for IE 😢 : https://github.com/ckeditor/ckeditor4-vue/pull/127/commits/9b6de22c96cd1a5759e2fe8787a549afb580ede4

Closes #124

jacekbogdanski commented on tests/component.js:
```diff
@@ -396,8 +396,18 @@ describe( 'CKEditor Component', () => {
 	function createComponent( props ) {
+		const fakeParent = window.document.createElement( 'span' );
+
+		props = {...props}
```
```suggestion
		props = { ...props }
```

jacekbogdanski commented on tests/integration.js:
```diff
@@ -97,15 +87,27 @@ describe( 'Integration of CKEditor component', () => {
+
+		deleteCkeditorScripts();
+
+		return createComponent( { editorUrl: basePath + 'ckeditor.js' } ).then( (comp) => {
```
```suggestion
		return createComponent( { editorUrl: basePath + 'ckeditor.js' } ).then( ( comp ) => {
```

jacekbogdanski commented on tests/integration.js:
```diff
@@ -139,4 +144,14 @@ describe( 'Integration of CKEditor component', () => {
+		const scripts = Array.from( document.querySelectorAll( 'script' ) );
+		const ckeditorScripts = scripts.filter( scriptElement => {
+			return scriptElement.src.indexOf( 'ckeditor.js' ) > -1;
+		});
```
```suggestion
		} );
```

jacekbogdanski reviewed (changes requested):
I didn't notice it earlier but it looks like we lost 100% test coverage somehow. Could you take a look @sculpt0r ?
![image](https://user-images.githubusercontent.com/10279586/145050561-263fb619-e9b5-490e-9a2f-ba062f9350aa.png)

sculpt0r commented:
The gap I coverage appears due to the fact - that now there is no editor created with empty config. Because I added `observableParent` to each config. I'll figure it out.

sculpt0r commented:
@jacekbogdanski 
- adjusted tests so after each test namespace clean up is made
- added test for creating editor with the default config. I put it at the end - so no other tests fail because of it. It doesn't look well... I left a comment above the test. However with #102 the tests will be touched again - so maybe it will be a good time to look at the tests with config again.

jacekbogdanski reviewed (approved):
Looks good 👍🏻 We can handle tests related to delayed initialization in the dedicated PR.
